### PR TITLE
Add more logging in case events are not contiguous

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Add more logging including namespace ID & workflow ID & run ID if DB encounters data loss

**Why?**
N/A

**How did you test it?**
N/A

**Potential risks**
N/A

**Is hotfix candidate?**
No

### CODE REVIEW

File: service/frontend/workflowHandler.go [hunk 1]
```diff
@@ -3312,7 +3313,14 @@ func (wh *WorkflowHandler) getHistory(
...
 	})
-	if err != nil {
+	switch err.(type) {
+	case nil:
+		// noop
+	case *serviceerror.DataLoss:
+		// log event
+		wh.GetLogger().Error("encounter data loss event", tag.WorkflowNamespaceID(namespaceID), tag.WorkflowID(execution.GetWorkflowId()), tag.WorkflowRunID(execution.GetRunId()))
+		return nil, nil, err
+	default:
```
Comment (alexshtin) [inline service/frontend/workflowHandler.go:3323]: So the canonical way to do this is:
```suggestion
	var dataLossErr *serviceerror.DataLoss
	if errors.As(err, &dataLossErr) {
		// log event
		wh.GetLogger().Error("encounter data loss event", tag.WorkflowNamespaceID(namespaceID), tag.WorkflowID(execution.GetWorkflowId()), tag.WorkflowRunID(execution.GetRunId()))
		return nil, nil, err
	}
	if err != nil {
```

File: common/persistence/historyStore.go [hunk 2]
```diff
@@ -456,13 +456,13 @@ func (m *historyV2ManagerImpl) readHistoryBranch(
 			// In that case we don't validate history continuousness for the first page
 			// TODO: in this case, some events returned can be invalid(stale). application layer need to make sure it won't make any problems to XDC
 			if defaultLastEventID == 0 || token.LastEventID != defaultLastEventID {
-				logger.Error("Corrupted non-continuous event batch",
+				logger.Error("Corrupted non-contiguous event batch",
```
Comment (alexshtin) [inline common/persistence/historyStore.go:459]: This is a good rule: "log error or return it but not both". I know there is no such thing as "too many logs" but still.

File: common/persistence/historyStore.go [hunk 3]
```diff
@@ -456,13 +456,13 @@ func (m *historyV2ManagerImpl) readHistoryBranch(
...
 					tag.FirstEventVersion(firstEvent.GetVersion()), tag.WorkflowFirstEventID(firstEvent.GetEventId()),
 					tag.LastEventVersion(lastEvent.GetVersion()), tag.WorkflowNextEventID(lastEvent.GetEventId()),
 					tag.TokenLastEventVersion(token.LastEventVersion), tag.TokenLastEventID(token.LastEventID),
 					tag.Counter(eventCount))
 				return historyEvents, historyEventBatches, nil, dataSize, lastFirstEventID, token.LastEventID,
-					serviceerror.NewInternal(fmt.Sprintf("corrupted history event batch, eventID is not continuous"))
+					serviceerror.NewDataLoss(fmt.Sprintf("corrupted history event batch, eventID is not contiguous"))
```
Comment (alexshtin) [inline common/persistence/historyStore.go:465]: Remove `fmt.Sprintf` or add eventIds to the message.

**Review by alexshtin — APPROVED** [review]

File: common/persistence/historyStore.go (hunk 2 above)
Comment (wxing1292) [reply common/persistence/historyStore.go:459]: i agree, but this situation is different, this layer has enough info about tree id, branch id, event id.
new logs try to add more details about workflow / run id

File: common/persistence/historyStore.go (hunk 3 above)
Comment (wxing1292) [reply common/persistence/historyStore.go:465]: the fmt sprintf should be changed during the grpc migration

File: service/frontend/workflowHandler.go (hunk 1 above)
Comment (wxing1292) [reply service/frontend/workflowHandler.go:3323]: let us talk, i believe the solution you are suggesting is even verbose

**Pull request merged**